staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -120,6 +120,37 @@ type Config struct {
 	// Version string
 }
 
+var _ fmt.Stringer = new(Config)
+var _ fmt.GoStringer = new(Config)
+
+// GoString implements fmt.GoStringer and sanitizes sensitive fields of Config
+// to prevent accidental leaking via logs.
+func (c *Config) GoString() string {
+	return c.String()
+}
+
+// String implements fmt.Stringer and sanitizes sensitive fields of Config to
+// prevent accidental leaking via logs.
+func (c *Config) String() string {
+	if c == nil {
+		return "<nil>"
+	}
+	cc := CopyConfig(c)
+	// Explicitly mark non-empty credential fields as redacted.
+	if cc.Password != "" {
+		cc.Password = "--- REDACTED ---"
+	}
+	if cc.BearerToken != "" {
+		cc.BearerToken = "--- REDACTED ---"
+	}
+
+	// Note: below format string skips Impersonate, AuthProvider,
```

Comment:
Is `Impersonate` sensitive?  I could see an argument for `Extra`...

Reply:
It is according to `AnonymousClientConfig`: https://github.com/kubernetes/kubernetes/blob/f9b6441037a5e89287a0a123d338416cc8c9d13d/staging/src/k8s.io/client-go/rest/config_test.go#L265-L277

Reply:
Impersonate doesn't carry credentials, it is security-related in the same way username is. We should not omit it when logging